duos_ui/lib/screens/chat_page.dart

```diff
@@ -1,201 +1,195 @@
+class ChatPage extends StatefulWidget {
+  const ChatPage({Key? key, required this.arguments}) : super(key: key);
+
+  final ChatPageArguments arguments;
 
   @override
-  Widget build(BuildContext context) {
-    return ListView(
-      children: [
-        _DateLabel(label: "Yesterday"),
-        _MessageTile(message: "Hey how is it going"),
-        OwnMessageTile(message: "how is your day"),
-        _MessageTile(message: "it is good"),
-        _MessageTile(message: "how about you"),
-        OwnMessageTile(message: "it could be better"),
-      ],
-    );
-  }
+  ChatPageState createState() => ChatPageState();
 }
 
-class _DateLabel extends StatelessWidget {
-  const _DateLabel({
-    Key? key,
-    required this.label,
-  }) : super(key: key);
+class ChatPageState extends State<ChatPage> {
+  late String uid1;
+  late String uid2;
+  List<QueryDocumentSnapshot> listMessage = [];
+  int limit = 25;
 
-  final String label;
+  TextEditingController textEditingController = TextEditingController();
+  ScrollController listScrollController = ScrollController();
+
+  late ChatProvider chatProvider;
+  late AuthProvider authProvider;
 
   @override
-  Widget build(BuildContext context) {
-    return Center(
-      child: Padding(
-        padding: const EdgeInsets.symmetric(vertical: 32.0),
-        child: Text(label),
-      ),
-    );
+  void initState() {
+    super.initState();
+    chatProvider = context.read<ChatProvider>();
+    authProvider = context.read<AuthProvider>();
+
+    readLocal();
   }
-}
 
-class _MessageTile extends StatelessWidget {
-  const _MessageTile({
-    Key? key,
-    required this.message,
-  }) : super(key: key);
 
-  final String message;
+  void readLocal() {
+    // uid1 = authProvider.sub;
+    // uid2 = widget.arguments.peerUid;
+    uid1 = "PbsHaVsmcZQtspSJvAPlzgCPmP72";
+    uid2 = "zmUYi77QgaSPX6lmSSx0LlT67EV2";
+  }
 
   @override
   Widget build(BuildContext context) {
-    return Padding(
-      padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 4),
-      child: Align(
-        alignment: Alignment.centerLeft,
-        child: Column(
-          mainAxisSize: MainAxisSize.min,
-          crossAxisAlignment: CrossAxisAlignment.start,
-          children: [
-            Container(
-              decoration: BoxDecoration(
-                color: Colors.lightBlue[50],
-                borderRadius: const BorderRadius.only(
-                  topLeft: Radius.circular(18),
-                  topRight: Radius.circular(18),
-                  bottomRight: Radius.circular(18),
-                ),
+    return MaterialApp(
+      home: Scaffold(
+        appBar: AppBar(
+          backgroundColor: Colors.transparent,
+          centerTitle: true,
+          title: Row(
+            children: const [
+              Padding(
+                padding: EdgeInsets.only(right: 12, left: 20),
+                child: Avatar.small(url: "https://picsum.photos/200/300"),
```

Comment:
another hardcoded value picture?

Reply:
will change once contact page is working